### PR TITLE
OIDC login

### PR DESCRIPTION
**Pull request by daikema**
This is intended to be paired with / in support of https://github.com/CARTAvis/carta-controller/pull/119 where I've tried to elaborate a little bit on the changes made.

My guess is that for the most part it's probably useful to try to consolidate most of the discussion regarding the changes there as that's where most of them were made.

### CODE REVIEW

File: src/services/ApiService.ts [hunk 1]
```diff
@@ -183,16 +219,30 @@ export class ApiService {
...
+        } else if (ApiService.RuntimeConfig.oidcClientId) {
+            let usp = new URLSearchParams();
+            usp.set("id_token_hint", localStorage.getItem("oidc_id_token"));
+            usp.set("post_logout_redirect_uri", window.location.href);
+
+            localStorage.removeItem("oidc_refresh_token");
+            localStorage.removeItem("oidc_id_token");
+            localStorage.removeItem("authenticationType");
+
+            window.open(ApiService.RuntimeConfig.logoutAddress + "?" + usp.toString(), "_self");
```
Comment (veggiesaurus) [inline src/services/ApiService.ts]: I'd prefer `${ApiService.RuntimeConfig.logoutAddress}?${usp.toString()}` but I don't have a strong opinion about it :slightly_smiling_face:

File: src/services/ApiService.ts [hunk 2]
```diff
@@ -158,6 +160,40 @@ export class ApiService {
             } catch (e) {
                 return false;
             }
+        } else if (ApiService.RuntimeConfig.oidcClientId) {
+            try {
+                if (!this.oidcAxiosInstance) {
+                    //console.log("Creating OIDC Axios instance")
```
Comment (YuHsuan-Hwang) [inline src/services/ApiService.ts:166]: Please remove comment.

I wonder what the changes in `package-lock.json` are for?

Comment (veggiesaurus) [reply src/services/ApiService.ts:166]: > I wonder what the changes in package-lock.json are for?

I was wondering that too :thinking:

**Comment by daikema** [comment]
As per suggestion from @veggiesaurus updated this to a draft as I'm working on redoing the login doing more on the controller side, which hopefully won't require any changes to the frontend code.  That's still work in progress though.

**Comment by YuHsuan-Hwang** [comment]
Implemented in #2016.

**Pull request closed without merging**